node-operators/hyperdrive/disaster-recovery-and-node-migration.md

```diff
@@ -38,10 +38,60 @@ Wallet recovery is the process for restoring a node from an offline seed phrase
+First, [install Hyperdrive as usual](installation.md) on your new node. When Hyperdrive asks you to create a wallet, reply `n`. This will save you the step of creating and validating a new mnemonic which won't be used. 
 
 Next, use the `hyperdrive wallet recover` command to regenerate your node wallet based on your seed phrase backup. From here, you should review your configuration with `hyperdrive service config` and [monitor the node as usual](monitoring-your-node.md) to ensure it is operational.
 
+### Recovering the StakeWise Module
+
+After you've recovered your wallet, you can enable the StakeWise module in the Hyperdrive Configuration terminal `hyperdrive service config` -> Modules menu.
+
+After exiting the configuration terminal, Hyperdrive asks you to verify you don't have any running validators, and a final verification that if you did have validators running, you've waited 15 minutes, because Hyperdrive can't determine if you attested in the last 15 minutes. Because you're about to regenerate the same keys, ensure that the node you're recovering hasn't attested in the last 15 minutes.
+
+After you've enabled the StakeWise module, you need to regenerate your validator keys. The reason for this is that the new Hyperdrive installation currently has no way of knowing how many keys were generated before. Regenerate the same number of keys with `hyperdrive stakewise wallet generate -c X` where `X` is the number of keys to generate. The alias version of this command is `hyperdrive sw w g -c X`.
+
+Hyperdrive will ask if you want to continue uploading your deposit data. Even though it's already been uploaded, it's safe to do this, as Hyperdrive will detect they've already been uploaded, and report back the status.
+
+```
+NOTE: There is currently no way to remove a validator's deposit data from the NodeSet service once you've uploaded it. The key will be eligible for activation at any time, so this node must remain online at all times to handle activation and validation duties.
+If you turn the node off, you may be removed from NodeSet for negligence of duty!
+
+Do you want to continue uploading your deposit data? [y/n]
+y
+
+Uploading deposit data to the NodeSet server...
+All of your validator keys are already registered (10 in total).
+7 are pending activation.
+3 have been activated already.
+```
+
+### Recovering the Constellation Module
+
+After you've recovered your wallet, you can enable the Constellation module in the Hyperdrive Configuration terminal `hyperdrive service config` -> Modules menu.
+
+After exiting the configuration terminal, Hyperdrive asks you to verify you don't have any running validators, and a final verification that if you did have validators running, you've waited 15 minutes, because Hyperdrive can't determine if you attested in the last 15 minutes. Because you're about to regenerate the same keys, ensure that the node you're recovering hasn't attested in the last 15 minutes.
```

Comment:
```suggestion
After exiting the configuration terminal, Hyperdrive asks you to verify you don't have any running validators, and a final verification that if you did have validators running, you've waited 15 minutes, because Hyperdrive can't determine if you attested in the last 15 minutes. Because you're about to regenerate the same keys, this will apply to you, and **you MUST ensure that the node you're recovering hasn't attested in the last 15 minutes or you may be slashed!**
```